### ISSUE TITLE
UnquotedResourceTitleCheck needs fixing for variable based titles

### ISSUE THREAD

**Issue by petems** [iwarapter/sonar-puppet#253]
Opened a ticket and talked to the docs team about this: https://tickets.puppetlabs.com/browse/DOCUMENT-468

tl;dr Unquoted title checks should only apply for non variable based titles.

``` puppet
file { foo:    # Noncompliant
  ensure => directory,
}

file { [ foo, bar]:    # Noncompliant
  ensure => directory,
}
```

However:

``` puppet
file { $foo:   # Compliant
  ensure => directory,
}

file { [$foo, $bar]: # Compliant
  ensure => directory,
}
```

**Comment by iwarapter** [comment]
Thanks for your pull request :)

**Comment by petems** [comment]
Damn, meant to put `closes #254` in the PR so this auto-closes, forgot! :smile:
